docs/proposals/loadbalancing.md

```diff
@@ -0,0 +1,338 @@
+             Path1      Path2                       Path3        Path4
+             type: l7   type: l7                    type: l4     type: l4
+             foo.com    bar.com                     foobar.com   baz.com
+             /a - svc1  /prod - svc2                cert: c1     cert: c2
+             /b - svc1                              svc4         svc5
+```
+
+#### Ingress Path
+
+This is an example ingress path. It encapsulates:
+* L7 proxying information (i.e map these urls to these backends)
+* Security configuration (i.e tlsMode, secret)
+* Metadata needed for more advanced loadbalancing (e.g session persistence)
+
+See [draft pr](https://github.com/kubernetes/kubernetes/pull/12825) for an actual description of the resource. See [alternatives](#alternatives) for why this deviates from existing implementations in OpenShift or GCE:
+
+```yaml
+apiVersion: v1
+kind: IngressPoint
+metadata:
+  name: l7ingress
+  type: l7
+spec:
+  host: example
+  tlsMode: Termination
+  secret: foosecret
+  pathMap:
+    "foo.example.com":
+    - url: "/foo/*"
+      service:
+        name: foosvc
+        namespace: default
+        port: 80
+    - url: "/bar/*"
+      service:
+        name: barsvc
+        namespace: default
+        port: 80
+    "bar.example.com"
+    - url: "/foobar/*"
+        service:
+          name: foobarsvc
+          namespace: default
+          port: 80
+status:
+  # include host here when we have DNS figured out.
+  address: 104.101.11.39
+```
+
+Put more rigidly:
+
+```go
+// IngressPoint represents a point for ingress traffic.
+type IngressPoint struct {
+	TypeMeta
+	ObjectMeta
+	Spec IngressPointSpec
+	Status IngressPointStatus
+}
+
+// IngressPointSpec describes the ingressPoint the user wishes to exist.
+type IngressPointSpec struct {
+	Host    string
+	IngressPoint map[string][]Path
+}
+
+// IngressPointStatus describes the current state of an ingressPoint.
+type IngressPointStatus struct {
+	Address string
+}
+
+// ServiceRef is a reference to a single service:port.
+type ServiceRef struct {
+	Name      string
+	Namespace string
+	Port      int64
+}
+
+// Path connects a url path regex to a service:port.
+type Path struct {
+	Url     string
+	Service ServiceRef
+}
+```
+
+
+#### Claims
+
+In the context of 1.1 a cluster will only have one l7 loadbalancer controller that claims all IngressPoints so we don't need claims. In the larger scheme of things, we might still not need claims, however they solve the impedance mismatch problem (why do I need a loadbalancer to expose my service?) because that is how you get an ip. If you want to expose your service, you need a public ip for it. No matter where you're running you can get one by creating a claim. If you want something to use an ip you already have, create a claim for it.
+
+The downside is the user needs to create another resource. We can make this easier by defaulting claim-less ingress paths to a new claim.
```

Comment:
The claim is a name of the claim resource, it isn't in the current IngressPoint resource because I'm not planning to do it for 1.1. I want to be able to allocated different loadbalancers for different paths of the same hostname, so it needs some fleshing out after the initial design.